### DISCUSSION TITLE
Small Doubt with defining type for cells

### DISCUSSION THREAD

**Discussion by VADDIJASWANT** [clientIO/joint#2484] [Q&A] [answered]
### Introduction

Hello JointJS Community,

I am currently working on a project where I need to add custom SVGs to the paper. I have written some code to achieve this, and it works well for adding multiple different SVGs to the paper, rendering them, and exporting a graph in JSON format.

However, I have encountered an issue where all the SVG elements change to the last added SVG element under certain circumstances. I have been unable to pinpoint the specific situation causing this issue, but I suspect it might be due to the type being the same for all the custom SVG elements.

Here is the code snippet for reference:

```
joint.shapes.custom[item.name] = joint.dia.Element.define('SVGS', {
    customType:'SVGS',
    attrs: {
        svg: {
            width: "calc(w)",
            height: "calc(h)",
            fill: "#d3d3d3",
            'fill-opacity': 1,               
            stroke: '#000', 
            'stroke-opacity': 1, 
            'stroke-width': 1, 
        },
        label: {
            textAnchor: 'start',
            textVerticalAnchor: 'bottom',
            y: -12,
            "font-size": 12,
            fill: '#000',
        },
    }
}, {
    markup: util.svg`${svgString}`,
});
```
can anyone please tell me if we can have same **(type)** for multiple elements with different markup? and I would also appreciate any insights or suggestions to help resolve this issue. 

Thank you in advance for your assistance.

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment, marked as answer]
You can have multiple elements of given type and each of them can have a different markup. The `markup` has to be set as an attribute, not on prototype.

```js
const el1 = MyElement();
const el2 = MyElement();
el1.set('markup', util.svg`${svgString1}`);
el2.set('markup', util.svg`${svgString2}`);
```

**Reply by VADDIJASWANT** [reply]
Thanks a lot for suggestion
